lib/Sema/TypeCheckConcurrency.h

```diff
@@ -380,19 +385,23 @@ namespace detail {
 /// Diagnose any non-Sendable types that occur within the given type, using
 /// the given diagnostic.
 ///
+/// \param topLevelLoc provides an alternate source location to be used for the
+/// main sendability diagnostic, but not for associated notes
+///
 /// \returns \c true if any errors were produced, \c false if no diagnostics or
 /// only warnings and notes were produced.
 template<typename ...DiagArgs>
 bool diagnoseNonSendableTypes(
-    Type type, SendableCheckContext fromContext, SourceLoc loc,
+    Type type, SendableCheckContext fromContext,
+    SourceLoc loc, SourceLoc topLevelLoc,
```

Comment:
So these two loc's and their meaning is a bit confusing. Based on the implementation, it appears that now the `loc` parameter corresponds to the location where this non-sendable `type` appears in the source. That location is used to point out what about the type isn't Sendable. 

This new `topLevelLoc` is now the location where the the provided diagnostic should be emitted.

Overall I think we don't want to add a 2nd parameter to this function, and instead add a new overload that takes two source locations, and have the one taking only one source location forward to the two-location one. This will cut-down on all the other changes where we're just duplicating the argument now, i.e., it'll elminate `component.getLoc(), component.getLoc(),` changes.